CDTDatastore/CDTReplicator/CDTReplicator.m

```diff
@@ -168,7 +168,11 @@ - (BOOL)startWithTaskGroup:(dispatch_group_t)taskGroup error:(NSError *__autorel
         }
 
         self.started = YES;
-        // retain self to make the replicator fire and forget.
+        
+        // In order for the replicator to be "fire and forget" we need to create a retain cycle,
+        // the retain cycle needs to be ended when the replicator stops, whether that is from a
+        // successful run or not. The point where this should be released is in the
+        // -replicatorStopped: method.
```

Comment:
I think you need to describe this at the class level -- broadly cover the lifecycle of the replicator and when retain/releases happen. Right now someone will only see this if they are working in the particular code area that actually does a retain/release -- these comments are definitely worthwhile, but I think there's something higher level required here at the class level.